compiler/rustc_data_structures/src/stack.rs

```diff
@@ -5,7 +5,7 @@ const RED_ZONE: usize = 100 * 1024; // 100k
 
 // Only the first stack that is pushed, grows exponentially (2^n * STACK_PER_RECURSION) from then
 // on. This flag has performance relevant characteristics. Don't set it too high.
-const STACK_PER_RECURSION: usize = 1 * 1024 * 1024; // 1MB
+const STACK_PER_RECURSION: usize = 1024 * 1024; // 1MB
```

Comment:
TBH, I think clippy is wrong to complain about this.  Especially in a `const`, there's absolutely nothing wrong with a `1 *`, and representing "X MB" as `X * 1024 * 1024` is *fine*.

Maybe use `1 << 20`?  That's also clearly `1 MiB`, and hopefully clippy wouldn't whine about it.

Reply:
I wish I agreed `1 << 20` was obviously 1 MB. Maybe I'm just uncommonly bad at all these bit twiddling operations.

Reply:
(For context: `k` is a *decimal* shift by 3, `M` is a decimal shift by 6, etc.  The elementary school "move the decimal point" is, essentially, a shift.  So `M` ≡ 10³, aka "<<₁₀ 3".  And similarly `Mi` is 2²⁰, aka `<< 20`.  So "kebi", "mebi", "gibi", "tebi", etc are "<< 10", "<< 20", "<< 30", "<< 40", etc.)